### ISSUE TITLE
2D legend not updated if last plot is trajectory

### ISSUE THREAD

**Issue by Juanlu001** [poliastro/poliastro#349]
🐞 **Problem**

While running the [Jupiter notebook](), I noticed that this displays the correct legend:

```
op = OrbitPlotter()

op.plot(ss_e0, label="Earth")
op.plot_trajectory(ic1.sample(np.linspace(0 * u.deg, 180 * u.deg, 50))[-1], label="Inner Cruise 1")
#op.plot(ss_efly)
op.plot(ss_aph_post, label="Back to Earth");
```

![index](https://user-images.githubusercontent.com/316517/39050165-a8fcd2fa-44a4-11e8-9888-83c9d2c5d585.png)

but, if the last plot is just a trajectory, then the trajectories are not shown:

```
op = OrbitPlotter()

op.plot(ss_e0, label="Earth")
op.plot_trajectory(ic1.sample(np.linspace(0 * u.deg, 180 * u.deg, 50))[-1], label="Inner Cruise 1")
#op.plot(ss_efly)
op.plot_trajectory(ss_aph_post.sample(np.linspace(180 * u.deg, 360 * u.deg, 50))[-1], label="Back to Earth");
```

![index2](https://user-images.githubusercontent.com/316517/39050220-df94705c-44a4-11e8-9493-3ad86a4d6ade.png)

💡 **Possible solutions**

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

**Comment by shreyasbapat** [comment]
Maybe we should give out labels in plot_trajectory and not in plot. Then it would be easy to solve :)

**Comment by Juanlu001** [comment]
Can you elaborate? You mean setting the legend in plot_trajectory?

**Comment by shreyasbapat** [comment]
I am not too sure about it, but yes!

**Comment by shreyasbapat** [comment]
The link in the Jupiter Notebook leads nowhere...:/

**Comment by shreyasbapat** [comment]
Here I propose 2-3 solutions :grimacing: 
I tried out the first one ->
1) Make plot trajectory accept orbit as well as the CartesianRepresentation. Something like this: 
`    def plot_trajectory(self, trajectory, *, label=None, color=None):
        """Plots a precomputed trajectory.

        Parameters
        ----------
        trajectory : ~astropy.coordinates.CartesianRepresentation
            Trajectory to plot.
        """
        if isinstance(trajectory, Orbit):
            _, trajectory_f = trajectory.sample()
        else:
            trajectory_f = `trajectory`
`` 
Then set the legend in plot_trajectory only.
2) Somehow return epoch from sample() method. This will make us use the ``set_label`` directly

**Comment by Juanlu001** [comment]
Oops, you were right about the link! Anyway, this was closed in #352.
